client/router_update.go

```diff
@@ -521,7 +521,7 @@ func (cli *VanClient) RouterUpdateVersionInNamespace(ctx context.Context, hup bo
 			ownerRefs = []metav1.OwnerReference{*siteOwnerRef}
 		}
 		policyValidator := NewClusterPolicyValidator(cli)
-		for _, clusterRole := range ClusterRoles() {
+		for _, clusterRole := range cli.ClusterRoles(siteConfig.Spec.EnableClusterPermissions) {
 			// optional (in case of failure, cluster admin can add necessary cluster roles manually)
 			kube.CreateClusterRole(clusterRole, cli.KubeClient)
```

Comment:
Currently, if a specific site has been previously created and the user tries to enable all cluster permissions they won't be able, since the clusterRole is just created, not updated. Would this be an issue?

Reply:
I think that is fine. The update function is really only intended to update from an older version at present, not to try and adjust to changes in config (which we do aspire to, but should be automatic).